core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -545,14 +545,19 @@ object ConsumerGroupCommand extends Logging {
         .map(partitionInfo => new TopicPartition(topic, partitionInfo.partition))
     }
 
-    private def getPartitionsToReset(groupId: String): Iterable[TopicPartition] = {
+    private def getPartitionsToReset(groupId: String): Seq[TopicPartition] = {
+      val allTopicPartitions = adminClient.listGroupOffsets(groupId).keys.toSeq
       if (opts.options.has(opts.allTopicsOpt)) {
-        adminClient.listGroupOffsets(groupId).keys
+        allTopicPartitions
       } else if (opts.options.has(opts.topicOpt)) {
         val topics = opts.options.valuesOf(opts.topicOpt).asScala
-        parseTopicPartitionsToReset(topics)
+        val topicPartitions = parseTopicPartitionsToReset(topics)
+        allTopicPartitions.filter(topicPartition => topicPartitions.contains(topicPartition))
```

Comment:
I'm not sure I get why we filter based on current committed offsets. Why would we not permit the tool to set offsets for a topic partition which hasn't been previously consumed? If that is really the behavior we want, I would prefer to raise an exception rather than silently proceeding with only some of the offsets.